### PR TITLE
Shortlinks: Fix the UI in newer version of Gutenberg

### LINKED ISSUE #14982: Short link: buttons are off in Gutenberg 7.7
In Gutenberg 7.7, short link copy button is a little bit off visually:

Screenshot 2020-03-13 at 16 38 49

Testing with WordPress.com simple site with Varia theme.

#### Steps to reproduce the issue

1. Open Jetpack sidebar in the block editor, using Gutenberg 7.7
2. Observe the short link section's inputs

### PR DESCRIPTION
**Pull request by apeatling**
Fixes #14982

Before:
Screen Shot 2020-03-19 at 12 01 42 PM

After:
Screen Shot 2020-03-19 at 12 01 26 PM

#### Testing instructions:
* Make sure the Shortlinks module is active on your site.
* Publish a post on a publicly accessable site
* Hit the Jetpack logo in the editor
* Make sure the button and textbox line up
* Disable the Gutenberg plugin and make sure they still line up on the core WP version of Gutenberg.

#### Proposed changelog entry for your changes:

* None

### CODE REVIEW

File: extensions/shared/clipboard-input.scss
Comment (simison) [inline extensions/shared/clipboard-input.scss]: Since this is actually a Gutenberg core variable (`$button-size`), let's pull it as a variable in already now — that way we'll avoid fixes like this in future when things just get updated automatically:

https://github.com/WordPress/gutenberg/blob/bbc2390d5bd48d754e3cc4033d972eb2aa725e2e/packages/base-styles/_variables.scss#L39

You can have those variables available here by importing this:
```scss
@import './styles/gutenberg-base-styles.scss';
```

Comment (simison) [reply extensions/shared/clipboard-input.scss]: Oh looks like it's actually pretty new value: https://github.com/WordPress/gutenberg/pull/20529 and isn't pushed to NPM yet. 😁 

So let's use the old `$icon-button-size` variable meanwhile? Jetpack has base-styles v1.0.0 (latest is v1.4.0) and both have that: 

https://unpkg.com/browse/@wordpress/base-styles@1.0.0/_variables.scss
https://unpkg.com/browse/@wordpress/base-styles@1.4.0/_variables.scss

The variable just gets updated whenever someone updates the package to the latest in Jetpack.

Comment (jeherve) [reply extensions/shared/clipboard-input.scss]: > The variable just gets updated whenever someone updates the package to the latest in Jetpack.

Noting that this is tracked (update to 1.4.0) in #13593

**Comment by apeatling** [comment]
@simison Thanks for pointing me to the variables, super useful to know about those.

File: extensions/shared/clipboard-input.scss
Comment (jeherve) [reply extensions/shared/clipboard-input.scss]: We're now running @wordpress/base-styles 1.4.0 as of #15059. You should be all set to use the variable now!

Comment (glendaviesnz) [reply extensions/shared/clipboard-input.scss]: I tried updating this and if I rebase to master and do an yarn install it shows @wordpress/base-styles  as 1.4.0 but _variables.scss still looks like the 1.0.0 versions - tried deleting node modules and yarn install again and go the same thing - any ideas why this might happen?

Comment (jeherve) [reply extensions/shared/clipboard-input.scss]: It looks like `$button-size` was added on March 2. in https://github.com/WordPress/gutenberg/commit/7866ae1b48d8c3fd3c898598b6eac18b9fc2abe9#diff-0e7bf8af1d20999faf370cb9696b8c02 , but 1.4.0 was tagged on February 10. I'm afraid we'll need to wait for another release.

Until then, we can do what @simison suggested above maybe?

> So let's use the old $icon-button-size variable meanwhile?

Comment (simison) [reply extensions/shared/clipboard-input.scss]: @gziolo would it be possible to tag latest version for `base-styles`? Would find it handy in Calypso too.

Comment (gziolo) [reply extensions/shared/clipboard-input.scss]: WordPress 5.4 is currently scheduled to land on March 31 2020 and it's today :)

We can publish the latest version of all packages soon after, I personally can tackle it tomorrow if everything goes smooth.

Comment (gziolo) [reply extensions/shared/clipboard-input.scss:6]: @jorgefilipecosta is handling the release later today.

Comment (scruffian) [reply extensions/shared/clipboard-input.scss:6]: I added a PR to update to the latest version of base styles here: https://github.com/Automattic/jetpack/pull/15280

Comment (jeherve) [reply extensions/shared/clipboard-input.scss]: done in ee61e1464f515cacbd88198949ffd3230392c664

**Review by jeherve — APPROVED** [review]
This appears to work well in my tests with WP 5.3.2, 5.4, and 5.4+Gutenberg. It should be good to merge.

I'd love to have another review on this though! :)

**Review by sergeymitr — APPROVED** [review]

**Comment by jeherve** [comment]
Cherry-picked to `branch-8.4` in 96649823d5a725eb1caea7615bd2944c46407b8b

**Comment by jeherve** [comment]
r205421-wpcom

**Pull request merged**
